File: app/src/main/kotlin/io/ahxxm/ic/domain/CompressionOptions.kt
package io.ahxxm.ic.domain

enum class Encoder {
    MOZJPEG,
    JPEGLI
}

data class CompressionOptions(
    val quality: Int = 85,
    val preserveExif: Boolean = true,
    val convertPng: Boolean = false,
    val encoder: Encoder = Encoder.MOZJPEG
)

File: ios/ic/ic/Domain/CompressionOptions.swift
enum QualityLevel: String, CaseIterable {
    case ok, good, great

    var label: String { rawValue }
    // mozjpeg quality values matching Android app
    var quality: Int {
        switch self {
        case .ok: 75
        case .good: 85
        case .great: 90
        }
    }
}

struct CompressionOptions: Hashable {
    let quality: Int
    let preserveExif: Bool
    let convertPng: Bool
}
